packages/react-dom/src/client/DOMPropertyOperations.js

```diff
@@ -149,9 +150,56 @@ export function setValueForProperty(
   if (shouldIgnoreAttribute(name, propertyInfo, isCustomComponentTag)) {
     return;
   }
+
+  if (
+    enableCustomElementPropertySupport &&
+    isCustomComponentTag &&
+    name[0] === 'o' &&
+    name[1] === 'n'
+  ) {
+    let eventName = name.replace(/Capture$/, '');
+    const useCapture = name !== eventName;
+    const nameLower = eventName.toLowerCase();
+    if (nameLower in node) {
+      eventName = nameLower;
+    }
+    eventName = eventName.slice(2);
+
+    const listenersObjName = eventName + (useCapture ? 'true' : 'false');
+    const alreadyHadListener =
+      (node: any)._listeners && (node: any)._listeners[listenersObjName];
+
+    if (typeof value === 'function' || alreadyHadListener) {
+      if (!(node: any)._listeners) {
+        (node: any)._listeners = {};
```

Comment:
I hid the existing discussion to clean it up. As discussed before, let's change `_listeners` to use a random suffix matching what we do for other fields in DOMComponentTree.

Then we'll look again at the other approach Seb suggested earlier.

Reply:
Done!
I added a note in my list of followups to consider Seb's approach in the other thread.